inf-clojure.el

```diff
@@ -920,16 +920,19 @@ Prefix argument AND-GO means switch to the Clojure buffer afterwards."
   "Insert FORM into process and evaluate.
 Indent FORM.  FORM is expected to have been trimmed."
   (let ((clojure-process (inf-clojure-proc)))
-    (with-current-buffer (process-buffer clojure-process)
-      (comint-goto-process-mark)
-      (let ((beginning (point)))
-        (insert (format "%s" form))
-        (let ((end (point)))
-          (goto-char beginning)
-          (indent-sexp end)
-          ;; font-lock the inserted code
-          (font-lock-ensure beginning end)))
-      (comint-send-input t t))))
+    (with-selected-window (or (get-buffer-window inf-clojure-buffer)
```

Comment:
What is this for?

Reply:
I linked the CIDER PR that does the same thing. That points to https://stackoverflow.com/questions/13530025/emacs-scroll-automatically-when-inserting-text . If the point is not visible the `(goto (point-max))` code won't actually scroll the buffer. 

This change ensures that inserting into the repl will scroll the repl so you can see what you inserted and the result. Before this it only worked if the point was at the end and visible.

Reply:
Got it - admittedly I didn't read the PRs description, only the title. :D Might be good to mention this in some comment as well.

Reply:
on it! thanks!